Docs/devguide.md

```diff
@@ -0,0 +1,301 @@
+# Developer Guidance
+
+## Directory Structure
+
+The code is organized into three sections:
+
+- **LoRaEngine** - a .NET Standard 2.0 solution with the following folders:
+  - **modules** - Azure IoT Edge modules.
+  - **LoraKeysManagerFacade** - An Azure function handling device provisioning (e.g. LoRa network join, OTAA) with Azure IoT Hub as persistence layer.
+  - **LoRaDevTools** - library for dev tools (git submodule)
+- **Arduino** - Examples and references for LoRa Arduino based devices.
+- **Template** - Contain code useful for the "deploy to Azure button"
+- **Samples** - Contains sample decoders
+- **Docs** - Additional modules, pictures and documentations
+
+## LoRaEngine
+
+A **.NET Core 2.1** solution with the following projects:
+
+- **modules** - Azure IoT Edge modules.
+  - **LoRaWanPktFwdModule** packages the network forwarder into an IoT Edge compatible docker container. See https://github.com/Lora-net/packet_forwarder and https://github.com/Lora-net/lora_gateway. If you are using a RAK833-USB see this [submodule](/Docs/LoRaWanPktFwdRAK833USB)
+  - **LoRaWanNetworkSrvModule** - is the LoRaWAN network server implementation.
+- **LoraKeysManagerFacade** - An Azure function handling device provisioning (e.g. LoRa network join, OTAA) with Azure IoT Hub as persistence layer.
+- **LoRaDevTools** - library for dev tools (git submodule)
+
+## The overall architecture
```

Comment:
Would it make sense to move this section to the architecutre document?

Reply:
I am referring to the "The overall architecture" chapter. Not well represented by the snipped above, sorry.

Reply:
I think this is good here for now as its a detailed diagram of the concept we lay down in the architecture doc. I have added a link to the dev guide in the architecture.md so anybody that requires low level details can refer here.

Reply:
Ok. Sounds good.